packages/contracts-bedrock/src/cannon/libraries/MIPSInstructions.sol

```diff
@@ -667,4 +667,69 @@ library MIPSInstructions {
             _cpu.nextPC = _cpu.nextPC + 4;
         }
     }
+
+    /// @notice Selects a subword of byteLength size contained in memWord based on the low-order bits of vaddr
+    /// @param _vaddr The virtual address of the the subword.
+    /// @param _memWord The full word to select a subword from.
+    /// @param _byteLength The size of the subword.
+    /// @param _signExtend Whether to sign extend the selected subwrod.
+    function selectSubWord(
+        uint32 _vaddr,
+        uint32 _memWord,
+        uint32 _byteLength,
+        bool _signExtend
+    )
+        internal
+        pure
+        returns (uint32 subword_)
+    {
+        (uint32 dataMask, uint32 bitOffset, uint32 bitLength) = calculateSubWordMaskAndOffset(_vaddr, _byteLength);
+        subword_ = (_memWord >> bitOffset) & dataMask;
+        if (_signExtend) {
+            subword_ = signExtend(subword_, bitLength);
+        }
+        return subword_;
+    }
+
+    /// @notice Returns a word that has been updated by the specified subword at bit positions determined by the virtual
+    /// address
+    /// @param _vaddr The virtual address of the subword.
+    /// @param _memWord The full word to update.
+    /// @param _byteLength The size of the subword.
+    /// @param _value The subword that updates _memWord.
+    function updateSubWord(
```

Comment:
Looks like the store ops still need to be updated to use this.

Reply:
Ack! Thanks for flagging

Reply:
Fixed.